generator/deploy_openshift_pod.py

```diff
@@ -0,0 +1,225 @@
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import logging
+import os
+import time
+import datetime
+from typing import Dict, List
+
+from kubernetes import config, client
+from kubernetes.client.rest import ApiException
+from generator.exceptions import GeneratorDeployException
+from generator.utils import run_command
+
+logger = logging.getLogger(__name__)
+
+
+class OpenshiftDeployer(object):
+    def __init__(
+        self,
+        volume_dir: str,
+        upstream_name: str,
+        project_name: str,
+        env_dict: Dict = None,
+    ):
+        """
+
+        :param volume_dir:
+        :param upstream_name:
+        :param project_name:
+        :param env_dict:
+        """
+        self.volume_dir = volume_dir
+        self.upstream_name = upstream_name
+        self.env_image_vars = env_dict
+        self.project_name = project_name
+        self.pod_name = None
+        self.api_token = None
+        self.pod_manifest = None
+
+    def get_pod_name_id(self):
+        timestamp = datetime.datetime.now().strftime("-%Y%M%d-%H%M%S%f")
+        return f"{self.upstream_name}-{timestamp}-deployment"
+
+    def create_pod_manifest(self):
+        self.pod_name = self.get_pod_name_id()
+        env_image_vars = OpenshiftDeployer.build_env_image_vars(self.env_image_vars)
+        volumeName = f"{self.project_name}-{self.upstream_name}"
+        self.pod_manifest = {
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {"name": self.pod_name},
+            "spec": {
+                "containers": [
+                    {
+                        "image": self.upstream_name,
+                        "name": self.upstream_name,
+                        "env": env_image_vars,
+                        "volumeMounts": [
+                            {"mountPath": self.volume_dir, "name": volumeName}
+                        ],
+                    }
+                ],
+                "restartPolicy": "Never",
+                "serviceAccountName": "packit-service",
+                "volumes": [
+                    {
+                        "name": volumeName,
```

Comment:
@phracek I can see that volumeName variable is dynamically generated, same for the claimName: does that actually work? is there some prior setup needed? (like creating the pool volumes)

Reply:
@TomasTomecek I see. Unfortunatelly not. in Betka's case `volumeDir` is `/tmp/betka-generator`. No dynamically generation is done. This case is the same. `volumeName` is used during container start.